Plugins/NativeEngine/Source/NativeEngine.cpp

```diff
@@ -1508,6 +1599,33 @@ namespace Babylon
         });
     }
 
+    void NativeEngine::SetStencil(const Napi::CallbackInfo& info)
+    {
+        const auto writeMask{info[0].As<Napi::Number>().Uint32Value()};
+        const auto stencilOpFail{info[1].As<Napi::Number>().Uint32Value()};
+        const auto depthOpFail{info[2].As<Napi::Number>().Uint32Value()};
+        const auto depthOpPass{info[3].As<Napi::Number>().Uint32Value()};
+        const auto func{info[4].As<Napi::Number>().Uint32Value()};
+        const auto ref{info[5].As<Napi::Number>().Uint32Value()};
+        
+        m_stencilState = BGFX_STENCIL_FUNC_RMASK(0xFF); //  always 0xFF
+        m_stencilState |= stencilOpFail;
+        m_stencilState |= depthOpFail;
+        // bgfx write mask is always 0xFF, to not change stencil value when writemask is 0
+        // its value is kept unchanged.
+        // https://github.com/bkaradzic/bgfx/blob/2c21f68998595fa388e25cb6527e82254d0e9bff/src/renderer_d3d11.cpp#L2874
```

Comment:
I don't understand this comment. Can you explain more?

Reply:
bgfx stencil write mask is always 0xFF. when bjs set write mask to 0x0, depth pass op changes to keep so stencil value is not modified....just like having a 0x00 write mask.